portal/src/app/components/all-experiments/Experiments.tsx

```diff
@@ -26,24 +26,41 @@ const DuplicateAriaLabel: string = ALL_EXPERIMENTS_TABLE_EN.TABLE_COLUMNS.LINKS.
 
 export const Experiments: React.FC = () => {
   const { testSuites, status }: IUseExperimentsData = useExperimentsData();
+  const navigate = useNavigate();
   const [openDeleteModal, setOpenDeleteModal] = useState<boolean>(false);
   const [checkedRows, setCheckedRows] = useState<Record<number, boolean>>({});
-  const experimentsData = useMemo(() => (testSuites ? parseExperimentsData(testSuites): []), [testSuites]);
-  const navigate = useNavigate();
-
+  const [experimentsData, setExperimentsData] = useState<ExperimentData[]>([]);
   
+  const [deleteHandled, setDeleteHandled] = useState(false);
```

Comment:
for what you need this state?

Reply:
We need it in order to run the useEffect only once and not infinite times

Reply:
you can handle all values that you want to delete and then is the deleted success you can filter all this values from `experimentsData` array

  const [deletedData, setDeletedData] = useState<ExperimentData[]>([]);
and use new `deletedData` state to filter values from the array

and you can remove the deleteHandled flag state

Reply:
You're right, that's a better writing.
Fixed :)